components/camera/replaypcd/replaypcd.go

```diff
@@ -124,12 +146,20 @@ func newPCDCamera(ctx context.Context, deps resource.Dependencies, conf resource
 
 // NextPointCloud returns the next point cloud retrieved from cloud storage based on the applied filter.
 func (replay *pcdCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
-	replay.mu.RLock()
-	defer replay.mu.RUnlock()
+	replay.mu.Lock()
+	defer replay.mu.Unlock()
```

Comment:
Ah, I suppose if NextPointCloud is expected to be called in serial, then we may as well use the write lock, so there's no need for a separate "downloading" cache.

Reply:
Checked with the Vision team to see if we have a convention for how lidars currently behave (we do not), so we can do what makes most sense for SLAM for now. Spoke with Esha offline and we she pointed out that NextPointCloud will be called serially once we make the correct modifications on the SLAM side. So we don't have to worry about the caller getting data out-of-order, however I do think we should keep the lock here for safety in case someone does call this in parallel (otherwise it might panic).